### PR TITLE
Fix keyboard nav when using a master detail grid

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail](https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail](https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail)

File: src/app/index.html [hunk 1]
```diff
@@ -11,4 +11,7 @@
   <li>
     <a skyAppLink="/visual/edit-in-modal-grid">Edit it modal grid</a>
   </li>
+  <li>
+    <a skyAppLink="/visual/master-detail-grid">Master/Detail Grid</a>
```
Comment (Blackbaud-AlexKingman) [inline src/app/index.html:15]: Looks like you might have forgot to check this file in? Getting a 404.

Comment (Blackbaud-TrevorBurch) [reply src/app/index.html]: Nope. That was removed due to the enterprise issues (as was discussed in flex time). I've fixed this.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail](https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail](https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail](https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail](https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail)

File: src/app/public/modules/ag-grid/ag-grid-wrapper.component.ts [hunk 2]
```diff
@@ -85,4 +88,20 @@ export class SkyAgGridWrapperComponent implements AfterContentInit {
...
     }
   }
+
+  private get isInEditMode(): boolean {
+    const primaryGridEditing = this.agGrid && this.agGrid.api && this.agGrid.api.getEditingCells().length > 0;
+    if (primaryGridEditing) {
+      return true;
+    } else {
+      let innerEditing: boolean = false;
+      this.agGrid.api.forEachDetailGridInfo((detailGrid: DetailGridInfo) => {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/ag-grid/ag-grid-wrapper.component.ts:98]: Looking at the logic branches, `this.agGrid` can still be undefined here and cause an error. Is there a better way to organize these conditionals to check for that?

Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/ag-grid/ag-grid-wrapper.component.ts]: I moved the checks (which are really sanity checks) just to give us strict null checks here.

**Review by Blackbaud-AlexKingman — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail](https://developer.blackbaud.com/skyux-ag-grid?_sp.spa=pr-fix-keyboard-nav-master-detail)

**Pull request merged**